champion.py

```diff
@@ -16,33 +15,41 @@ class BaseChampion:
         - auto attack
     """
 
-    def __init__(self, champion_name: str, item_names: Optional[List[str]] = None, level: int = 1):
+    def __init__(self, champion_name: str, inventory: Optional[List[BaseItem]] = None, level: int = 1):
         assert isinstance(level, int) and 1 <= level <= 18, "Champion level should be in the [1,18] range"
         self.level = level
-        self.inventory = Inventory(item_names=item_names)
-        self.orig_base_stats = self.get_champion_base_stats(ALL_CHAMPION_BASE_STATS[champion_name])
-        self.item_stats = self.inventory.get_items_total_stats(self.inventory.items)
+        self.orig_base_stats = sc.get_champion_base_stats(ALL_CHAMPION_BASE_STATS[champion_name].copy(), level=level)
+        if inventory is None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+        self.unique_item_passives = set()
+        for item in self.inventory:
+            if hasattr(item, "passive"):
+                self.apply_unique_item_passive(item)
```

Comment:
Another way is to make `apply_passive` a member of the item, which is `pass` by default and items with a passive just override it.